scanner/src/main/kotlin/ScanResultsStorage.kt

```diff
@@ -258,6 +262,32 @@ abstract class ScanResultsStorage {
         return result
     }
 
+    /**
+     * Read those [ScanResult]s for the given [packages] from the storage that are
+     * [compatible][ScannerCriteria.matches] with the provided [scannerCriteria]. Also, [Package.sourceArtifact],
+     * [Package.vcs], and [Package.vcsProcessed] are used to check if the scan result matches the expected source code
+     * location. That check is important to find the correct results when different revisions of a package using the
+     * same version name are used (e.g. multiple scans of a "1.0-SNAPSHOT" version during development). Return the list
+     * of [ScanResult]s mapped to the [Identifier]s of the [packages], wrapped in a [Result], which is a [Failure] if
+     * an unexpected error occurred and a [Success] otherwise.
+     */
+    fun read(packages: List<Package>, scannerCriteria: ScannerCriteria): Result<Map<Identifier, List<ScanResult>>> {
```

Comment:
Would it make sense to pass a Set<Package> rather than a list to prevent duplicated packages?

Reply:
I agree with the rationale, but I'd like to keep it because the related functions like `LocalScanner.scanPackages()` also work with a list of packages. If we want to address this, I think we should do it separately and also check the whole call chain, keeping in mind that ideally the order of packages is preserved.